### PR TITLE
network: implement CMDMempool command

### LINKED ISSUE #1074: Mempool P2P command handling support
We need to support CMDMempool handling.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #1074 .
CMDMempool returns hashes of transactions in mempool in chunks.

### CODE REVIEW

File: pkg/network/server.go [hunk 1]
```diff
@@ -517,6 +517,33 @@ func (s *Server) handleInvCmd(p Peer, inv *payload.Inventory) error {
 	return nil
 }
 
+// handleMempoolCmd handles getmempool command.
+func (s *Server) handleMempoolCmd(p Peer) error {
+	txs := s.chain.GetMemPool().GetVerifiedTransactions()
+	sz := len(txs)
+	if sz > payload.MaxHashesCount {
+		sz = payload.MaxHashesCount
```
Comment (roman-khimov) [inline pkg/network/server.go:525]: I don't think it's the way it should work. You're limiting your answer to `MaxHashesCount` transactions and then sending an `inv` for each transaction which doesn't make much sense. Instead you should be creating as much `inv` messages as needed with `MaxHashesCount` hashes inside each to transmit all hashes from the mempool.

Comment (fyrchik) [reply pkg/network/server.go]: No, I append hash of every transaction in the loop, sending if length exceeds `MaxHashesCount`. This if is purely for allocating precise amount of memory needed. I removed this as this case is rare and obviously hurt readability.

File: pkg/network/server.go [hunk 2]
```diff
@@ -517,6 +517,29 @@ func (s *Server) handleInvCmd(p Peer, inv *payload.Inventory) error {
...
+		hs = append(hs, txs[i].Hash())
+		if len(hs) < payload.MaxHashesCount && i != len(txs)-1 {
+			continue
+		}
+		inv := payload.NewInventory(payload.TXType, hs)
+		pkt, err := NewMessage(CMDInv, inv).Bytes()
+		if err != nil {
+			return err
+		}
+		err = p.EnqueueHPPacket(pkt)
```
Comment (roman-khimov) [inline pkg/network/server.go:534]: Why is it an HP packet?

Comment (fyrchik) [reply pkg/network/server.go]: Fixed. In C# it is `mempool` command itself which is HP.

**Pull request merged**